app/src/main/java/com/example/yard/Post.java

```diff
@@ -1,14 +1,145 @@
 package com.example.yard;
 
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
 import androidx.appcompat.app.AppCompatActivity;
-
+import androidx.core.content.FileProvider;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Build;
 import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.File;
 
+@RequiresApi(api = Build.VERSION_CODES.O)
 public class Post extends AppCompatActivity {
 
+    public static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 45;
+    public static final String TAG = "Post";
+    private Button takePicture;
```

Comment:
let the name reflecting their format. Changing from `takePicture` to `btnTakePicture` or `takePictureButton`. Similar to the rest of the fields.